### PR TITLE
fix/Added filter sorting for user with viewer role

### PR DESCRIPTION
**Pull request by mustafapc19**
Ref: #3137

Known issues:
- Filter/sort config is lost when we change view(i.e selecting another table)

### CODE REVIEW

File: packages/nc-gui-v2/composables/useViewFilters.ts [hunk 1]
```diff
@@ -60,18 +50,24 @@ export function useViewFilters(
...
       } else {
-        filters.value = await $api.dbTableFilter.read(view?.value?.id as string)
+        if (parentId) {
+          filters.value = await $api.dbTableFilter.childrenRead(parentId)
+        } else {
+          filters.value = await $api.dbTableFilter.read(view?.value?.id as string)
+        }
       }
+    } catch (e) {
+      console.log(e)
```
Comment (wingkwong) [inline packages/nc-gui-v2/composables/useViewFilters.ts:70]: `message.error(await extractSdkResponseErrorMsg(e))`

File: packages/nc-gui-v2/composables/useViewSorts.ts [hunk 2]
```diff
@@ -68,11 +54,16 @@ export function useViewSorts(
   }
 
   const deleteSort = async (sort: SortType, i: number) => {
-    if (isUIAllowed('sortSync') && sort.id && !isPublic.value && !isSharedBase) {
+    if (isUIAllowed('sortSync') && sort.id && !isPublic.value && !isSharedBase.value) {
       await $api.dbTableSort.delete(sort.id)
```
Comment (wingkwong) [inline packages/nc-gui-v2/composables/useViewSorts.ts:58]: add a try catch for api call. throw errors using `message.error(await extractSdkResponseErrorMsg(e))`

File: packages/nc-gui-v2/composables/useViewFilters.ts [hunk 3]
```diff
@@ -129,17 +125,11 @@ export function useViewFilters(
...
-    if (isPublic.value) {
-      filters.value[i] = { ...filter }
-
-      filters.value = [...filters.value]
-
-      return
-    }
     if (!view?.value) return
 
-    if (!isUIAllowed('filterSync')) {
```
Comment (wingkwong) [inline packages/nc-gui-v2/composables/useViewFilters.ts:141]: users without `filterSync` shouldn't be able to update the filters in meta.

File: packages/nc-gui-v2/composables/useViewData.ts [hunk 4]
```diff
@@ -84,7 +84,7 @@ export function useViewData(
   /** load row comments count */
   const loadAggCommentsCount = async () => {
     // todo: handle in public api
```
Comment (wingkwong) [inline packages/nc-gui-v2/composables/useViewData.ts:86]: have we handled that already? if so, remove this TODO tag.

File: packages/nc-gui-v2/composables/useViewFilters.ts (hunk 3 above)
Comment (mustafapc19) [reply packages/nc-gui-v2/composables/useViewFilters.ts:141]: @wingkwong `nestedMode` includes that

**Comment by dstala** [comment]
@mustafapc19 

[editor mode] 
- sort, filter worked fine
[commenter mode / viewer mode] 
- [ ] create filter, hard refresh - filter isn't retained
- [ ] reset filter requires hard refresh for spreadsheet to get updated
- [ ] apply sort, hard refresh - sort is lost

**Comment by dstala** [comment]
@mertmit mentioned that the reset filter issue is possibly fixed in the latest develop. rebase, and retry

**Review by wingkwong — APPROVED** [review]

**Review by dstala — APPROVED** [review]

**Pull request merged**